lib/gc/ExecutionEngine/OpenCLRuntime/OpenCLRuntimeWrappers.cpp

```diff
@@ -151,9 +160,16 @@ static cl_device_id getDevice(cl_device_type *devtype) {
       abort();
     }
 
-    std::vector<cl_device_id> devices(numDevices);
-    clGetDeviceIDs(platforms[i], *devtype, numDevices, devices.data(), nullptr);
-    return devices[0];
+    std::vector<cl_device_id> devices(uintValue);
+    clGetDeviceIDs(platform, *devtype, uintValue, devices.data(), nullptr);
+
+    for (auto &device : devices) {
+      CL_SAFE_CALL(clGetDeviceInfo(device, CL_DEVICE_VENDOR_ID, sizeof(cl_uint),
+                                   &uintValue, nullptr));
+      if (uintValue == 0x8086) { // Make sure this is an Intel device
```

Comment:
is there a enum or something else that contains vendor ids in opencl? I mean something like: `vendor_ids::Intel` instead of `0x8086`?

Reply:
I don't know any.